core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -177,6 +178,43 @@ public synchronized boolean isTerminated() {
   }
 
   /**
+   * Ensures that there is an active transport and that it is started.
+   */
+  public void tryStartTransport() {
+    obtainActiveTransport();
+  }
+
+  /**
+   * Shuts down the current transport and starts a new one. If there is no active transport, then
+   * starts one. This can be useful to force the transport to connect to a different backend when
+   * the remote endpoint is a load balancer.
+   */
+  public synchronized void recreateTransport() {
+    if (activeTransport != null) {
+      activeTransport.shutdown();
+      activeTransport = null;
+    }
+    tryStartTransport();
+  }
+
+  /**
+   * Pings the remote endpoint to verify that the transport is still active. The future will
+   * complete when the remote endpoint replies. The future completes exceptionally if the transport
+   * is shutdown before the response is received. The value of the future is the round-trip time of
+   * the ping, in nanoseconds.
+   *
+   * <p>Transports that do not have any mechanism by which to ping the remote endpoint should
+   * return {@link com.google.common.util.concurrent.Futures#immediateFuture(Object)
+   * immediateFuture(null)}.
+   *
+   * @return a future that completes when the remote endpoint responds to the ping with a value of
+   *     the round-trip time in nanoseconds
+   */
+  public ListenableFuture<Long> ping() {
```

Comment:
IMO, it's a transport operation. So it's useful here since this channel implementation is built on transports. But I'm not sure it makes sense on other channels. For example, I envision an in-process channel where ping has no value (or real meaning).

As far as Guava types, I can switch away from `ListenableFuture`. Eric Anderson and I exchanged a couple of messages about this: https://groups.google.com/d/topic/grpc-io/4Z8RbmQJ144/discussion

If you think we should pro-actively avoid Guava types in the public API, I'm happy to change it.